doc/dev/development.md

```diff
@@ -144,9 +144,7 @@ you may try relocating the repo to /var/tmp/ in order to build and install from
 
 - [Protocol Buffers](https://github.com/protocolbuffers/protobuf) v3.11.4. [Installation instructions](https://github.com/protocolbuffers/protobuf/blob/master/src/README.md).
 - [Protobuf-C](https://github.com/protobuf-c/protobuf-c) v1.3.3. [Installation instructions](https://github.com/protobuf-c/protobuf-c/blob/master/README.md).
-- gRPC plugin: [protoc-gen-go](https://github.com/golang/protobuf) v1.3.4. **Must match the proto version in
-src/control/go.mod.** Install the specific version using GIT_TAG instructions
-[here](https://github.com/golang/protobuf/blob/master/README.md).
+- gRPC plugin: [protoc-gen-go](https://github.com/golang/protobuf) v1.3.5. This plugin is automatically installed by the Makefile in $DAOSREPO/src/proto.
```

Comment:
I think it would be better to say it's the version is specified in go.mod rather than putting the current version number here. One less place to update it when it changes in the future.

Reply:
> > We shouldn't land it if you do not like it.
> 
> It's not that I don't like it. It's that I would be happier about landing it if there were a lot more tests demonstrating that it works now and will continue to work after future refactoring and other changes. Tests are also a good way to programmatically express requirements and explore possibilities before they're encountered in the real world.
> 
> > So let's discuss adding in an appropriate amount of automated test coverage. These are the highlights that are occurring in this PR. 1) detecting the network device class, 2) sharing that with the agent via the GetAttachInfo gRPC, 3) having the agent filter the devices added to the attach info cache based on the network device class provided, and 4) validating that each IO server instance is configured with, and starts with the same device class.
> > Do you have concerns with all 4 areas, or something in particular that would benefit from some unit testing?
> 
> 1. Is tricky, given that it requires interaction with the system unless you're using stubs, and then you're just testing the stubs. The best way to handle this kind of thing is to keep the untestable stuff as separate from the "business logic" as possible. In the case of the `netdetect` library, this might mean changing things around a bit to pass things like device affinity into `createFabricScanEntry()` so that a set of table-driven unit tests can explore possibilities with a variety of synthetic inputs. This might arguably be outside the scope of this PR and I'm not insisting on it. I would suggest opening another low-priority ticket to address it as a background task, though.
> 2. can be accomplished with a new `TestServer_MgmtSvc_GetAttachInfo` set of tests. I was going to suggest that you could just add on to what's already there, but I just looked and realized that no one ever added a set of test cases for that RPC handler. It's an ancient one that predates my time here. It could go alongside of `TestServer_MgmtSvc_LeaderQuery` and draw inspiration from it.
> 3. looks pretty straightforward, unless I'm missing something... You already have some test coverage for the attachInfoCache, so adding a scenario to verify that the filtering works as expected should be pretty easy.
> 
> For 4), you could add a test to `server/config_test.go` to validate that a config with two different interface types fails validation. You added the ability to stub out the real `GetDeviceClass()`, so this should be trivial. I don't think we need to go further than that.

I've added unit tests for 2,3,4.  I'll open a ticket to address the changes required for #1.

Reply:
> I think it would be better to say it's the version is specified in go.mod rather than putting the current version number here. One less place to update it when it changes in the future.

Done.